lib/fabrik/DirectorManager.js

```diff
@@ -211,6 +211,31 @@ class DirectorManager extends BaseManager {
                 previous_manifest: manifest
               }, opts.context);
               return;
+            })
+            .then(() => {
+              let updateActionContext = _.cloneDeep(actionContext);
+              _.chain(updateActionContext)
+                .set('instance_guid', this.getInstanceGuid(deploymentName))
+                .value();
+              _.chain(updateActionContext.params)
+                .set('service_id', this.service.id)
+                .set('plan_id', this.plan.id)
+                .value();
+              return this
+                .getDeploymentIps(deploymentName)
+                .then(ips => this.agent.preUpdate(ips, updateActionContext))
+                .tap(response => preUpdateResponse = response)
+                .catch(FeatureNotSupportedByAnyAgent, ServiceInstanceNotOperational, err => {
+                  logger.debug('+-> Caught expected error of feature \'preUpdate\':', err);
+                  return;
+                })
+                .catch(err => {
+                  if (err.status === CONST.HTTP_STATUS_CODE.NOT_FOUND) {
```

Comment:
if feature not supported first check in settings, rather than invoking and letting it error & then suppressing such error.

Reply:
We get list of supported features from `/info` end point of agent. Is there any way to get service specific supported features from settings also?